lib/Runtime/Library/JavascriptProxy.h

```diff
@@ -176,12 +176,14 @@ namespace Js
                 JavascriptConversion::ToPropertyKey(element, scriptContext, &propertyRecord, nullptr);
                 propertyId = propertyRecord->GetPropertyId();
 
-                if (!targetToTrapResultMap.ContainsKey(propertyId))
+                if (targetToTrapResultMap.ContainsKey(propertyId))
```

Comment:
Looks like AddNew returns -1 if the key is already present. Might we benefit from using it as a test-and-set operation rather than separate ContainsKey and Add steps?